tokio/src/process/windows.rs

```diff
@@ -167,28 +167,97 @@ unsafe extern "system" fn callback(ptr: PVOID, _timer_fired: BOOLEAN) {
     let _ = complete.take().unwrap().send(());
 }
 
-pub(crate) type ChildStdio = PollEvented<NamedPipe>;
+#[derive(Debug)]
+struct ArcFile(Arc<StdFile>);
 
-pub(super) fn stdio<T>(io: T) -> io::Result<PollEvented<NamedPipe>>
+impl io::Read for ArcFile {
+    fn read(&mut self, bytes: &mut [u8]) -> io::Result<usize> {
+        (&*self.0).read(bytes)
+    }
+}
+
+impl io::Write for ArcFile {
+    fn write(&mut self, bytes: &[u8]) -> io::Result<usize> {
+        (&*self.0).write(bytes)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        (&*self.0).flush()
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct ChildStdio {
+    // Used for accessing the raw handle, even if the io version is busy
+    raw: Arc<StdFile>,
+    // For doing I/O operations asynchronously
+    io: Blocking<ArcFile>,
+}
```

Comment:
It seems like you could drop the `Arc` stuff if you just stored the raw handle next to the `io` field.

Reply:
`std::process::Stdio` wants ownership of the handle (and will close it on drop) which might cause issues if the handle is still being written to in the background. Using the Arc allows us to potentially avoid a syscall to duplicate the handle if it doesn't have any pending I/O on it

Reply:
You could implement a `try_unwrap` on the `Blocking` type to get the same behavior without the `Arc`. But I don't mind either way, so I will approve it and let you decide if you want to implement that.

Reply:
I did consider adding `try_unwrap` to `Blocking`, the tricky bit is that it moves the handle to and from the blocking threadpool, so if there is a pending I/O operation we cannot access anything. The conversion from `ChildStd*` to `Stdio` are synchronous, so we cannot even internally `.await` things. Hence the `Arc` approach allows us to at least clone the handle if needed

I'm going to leave it as is for now since we can always revisit it later as needed. Thanks for the review!